File: Daily Challenge/2020/October/4_remove_covered_intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        if not intervals:
            return 0
        a = sorted(intervals, key=lambda x: x[0])
        to_delete = []
        for i in range(len(a)-1):
            if a[i][0] == a[i+1][0] and a[i][1] <= a[i+1][1]:
                to_delete.append(i)
        for i in to_delete:
            del a[i]
        b = [a[i][1] for i in range(len(a))]
        lMax = [b[0]]*len(b)
        res = []
        for i in range(1, len(b)):
            lMax[i] = max(b[i], lMax[i-1])
            if b[i] <= lMax[i-1]:
                res.append(i)
        return len(a)-len(res)

File: Daily Challenge/2020/October/4_remove_covered_intervals.cpp
bool compare(const vector<int> &a, const vector<int> &b){
    if(a[0] == b[0]) return a[1] > b[1];
    return a[0] < b[0];
}
class Solution {
public:
    
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), compare);
        vector<int> interval = intervals[0];
        int count = 1;
        int i = 1;
        while(i < intervals.size()){
            while(i < intervals.size() && intervals[i][0] >= interval[0] && intervals[i][1] <= interval[1])
                i++;
            if(i == intervals.size()) break;
            count++;
            interval = intervals[i];
            i++;
        }
        return count;
    }
};